Micronaut cloud database guide base and Oracle Cloud

burtbeckwith opened the pull request.

burtbeckwith committed:
allow apps to have no features

burtbeckwith committed:
micronaut-cloud-database base guide and Oracle Cloud guide

burtbeckwith commented:
I left some TODOs that I'll get back to in a bit and there's still some rework needed

burtbeckwith commented:
Here's a PDF of the guide: [micronaut-cloud-database-oracle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/8003990/micronaut-cloud-database-oracle.pdf)

sdelamo commented on guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc:
```diff
@@ -0,0 +1,386 @@
+external:micronaut-cloud-database-base/start.adoc[]
+
+* An Oracle Cloud account (create a free trial account at https://signup.oraclecloud.com[signup.oraclecloud.com])
```
I think this paragraph should be in a common snippet.

sdelamo commented on guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc:
```diff
@@ -0,0 +1,386 @@
+
+Use the guideLink:micronaut-oracle-cloud[Deploy a Micronaut application to Oracle Cloud] guide to create a compute instance and deploy the application to it; follow the steps in the "Create an Oracle Cloud Compute Instance" and "Deploy to Oracle Cloud" sections up to the step where you start the application. We need to connect the application to the MySQL database before starting it up.
+
+TODO ingress rule
```
Pending

sdelamo commented on guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc:
```diff
@@ -0,0 +1,386 @@
+export JDBC_PASSWORD=<user password>
+----
+
+TODO curl examples to cloud compute instance
```
Pending

sdelamo commented on guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc:
```diff
@@ -0,0 +1,386 @@
+}
+----
+
+external:micronaut-cloud-database-base/end.adoc[]
```
I don't understand the benefit of using this instead of a common snippet. Why did we needed `external`. Could not we just have?

```
common:micronaut-cloud-database-end.adoc

```

sdelamo replied:
Resolved via https://github.com/micronaut-projects/micronaut-guides/pull/743

sdelamo commented on guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc:
```diff
@@ -0,0 +1,386 @@
+"port-x": 33060
+----
+
+external:micronaut-cloud-database-base/create-app.adoc[]
```
I don't understand the benefit of using this instead of a common snippet. Why did we needed `external`. Could not we just have?

```
common:micronaut-cloud-database-create-app.adoc[]
```

sdelamo replied:
Resolved via https://github.com/micronaut-projects/micronaut-guides/pull/743

sdelamo commented on guides/micronaut-cloud-database-base/create-app.adoc:
```diff
@@ -0,0 +1,3 @@
+== Creating the Application
```
Could not we just have this in a common snippet name `micronaut-cloud-database-create-app`?

sdelamo commented on guides/micronaut-cloud-database-base/end.adoc:
```diff
@@ -0,0 +1 @@
+Read more about https://micronaut-projects.github.io/micronaut-data/latest/guide/[Micronaut Data] and https://micronaut-projects.github.io/micronaut-oracle-cloud/latest/guide/[Micronaut Oracle Cloud support].
```
Could not we just have this in a common snippet name `micronaut-cloud-database-end`?

kroegerk replied:
```suggestion
Read more about https://micronaut-projects.github.io/micronaut-data/latest/guide/[Micronaut Data] and https://micronaut-projects.github.io/micronaut-oracle-cloud/latest/guide/[Micronaut Oracle Cloud Support].
```

Title case

sdelamo commented:
I've added the PR to the project. Please, let's use the Github project.

burtbeckwith committed:
Merge branch 'master' into micronaut-cloud-database

burtbeckwith committed:
remove slug and asciidoctor

burtbeckwith committed:
add ingress rule step and curl examples

burtbeckwith committed:
only check output dir exists once

burtbeckwith committed:
updated apps merge logic

retain apps that are only in the base metadata or only in guide metadata, then merge features from apps in both

sdelamo commented on buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy:
```diff
@@ -336,14 +336,26 @@ class GuideProjectGenerator implements AutoCloseable {
-                app.features += baseAppMatchingName.features
-            }
+
+        Map<String, App> baseApps = base.apps.collectEntries { [(it.name): it] }
```
Can you please add this change to a separate PR. So that we can merge it as soon as possible?

burtbeckwith replied:
https://github.com/micronaut-projects/micronaut-guides/pull/767

sdelamo commented:
**WARNING: We cannot merge this until this PR is merged https://github.com/micronaut-projects/micronaut-guides/pull/685**

sdelamo commented:
@kroegerk can you proofread this guide? Thanks!

burtbeckwith committed:
Merge branch 'master' into micronaut-cloud-database

burtbeckwith committed:
delete .gitkeep

burtbeckwith committed:
PR feedback from Hari

highlight that there's no free-tier MySQL
add PowerShell cli info

graemerocher commented on guides/micronaut-cloud-database-oracle/metadata.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "base": "micronaut-cloud-database-base",
+  "title": "Deploy a Micronaut database application to Oracle Cloud",
```
Perhaps rename to `"Deploy a Micronaut MySQL database application to Oracle Cloud"` to clarify this is MySQL service

kroegerk replied:
```suggestion
  "title": "Deploy a Micronaut Database Application to Oracle Cloud",
```

Title case

burtbeckwith committed:
remove Oracle Cloud from shared

burtbeckwith committed:
Proofreading fixes

burtbeckwith committed:
update publication date
